skimage/util/compare.py

```diff
@@ -1,27 +1,52 @@
+import warnings
+from itertools import product
+
 import numpy as np
 
 from .dtype import img_as_float
-from itertools import product
+from skimage._shared.utils import (
+    DEPRECATED,
+)
 
 
-def compare_images(image1, image2, method='diff', *, n_tiles=(8, 8)):
+def compare_images(
+    image0=None, image1=None, image2=DEPRECATED, *, method='diff', n_tiles=(8, 8)
```

Comment:
While using `image0=None` has the advantage of being simple, I'd prefer a wrapper around `compare_images` here. That way, we can still perform the necessary checks, but `functools.wraps(func)` should ensure that IDE's and code completion pick up the actual signature I'm suggesting below
```suggestion
    image0, image1, image2=DEPRECATED, method='diff', *, n_tiles=(8, 8)
```
Note you can't yet move the `*` forward because we still want the old API to work during the deprecation cycle.

Reply:
Ok. Should I write a custom wrapper then? Just to be sure, do you agree that a) I can't use `deprecate_parameter` here and b) existing code like `compare_images(image1=a, image2=b)` should still work during the deprecation cycle as well?

Reply:
a) yes and b) yes. I think a specific wrapper should be able to address all concerns in this case. I think something like
```python

def _rename_image_params(func):

    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        # check / fix kwargs and warn if `image2` is used
        return func(*args, **kwargs)

    return wrapper
``` 
should work.

Reply:
Thank you, @lagru. I'll update the function and tests accordingly.